amd_tressfx_vulkan/src/TressFXSimulationVulkan.h

```diff
@@ -0,0 +1,133 @@
+//--------------------------------------------------------------------------------------
+// File: TressFXSimulation.h
+//
+// Hair physics simulation header
+//
+//
+// Copyright (c) 2016 Advanced Micro Devices, Inc. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//--------------------------------------------------------------------------------------
+
+#pragma once
+
+#include "AMD_TressFX.h"
+#include "Math\vector3D.h"
+#include "TressFXMeshVulkan.h"
+#include "Util.h"
+#include <map>
+#include <string>
+
+typedef DirectX::XMFLOAT4 float4;
+typedef DirectX::XMFLOAT3 float3;
+
+namespace AMD
+{
+
+class TressFXSimulation
+{
+  public:
+    TressFXSimulation(void);
+    ~TressFXSimulation(void);
+
+    VkResult OnCreateDevice(VkDevice pvkDevice, TressFX_CollisionCapsule *pCollision,
+                            uint32_t maxUniformCount, uint32_t cpu_memory_index,
+                            uint32_t gpu_memory_index);
+    VkResult Simulate(VkDevice pvkContext, VkCommandBuffer commandBuffer,
+                      float fElapsedTime, float density, tressfx_vec3 &windDir,
+                      float windMag, DirectX::XMMATRIX *pModelTransformForHead,
+                      ID3D11UnorderedAccessView *pSkinningTransforms,
```

Comment:
Mixing D3D11 and Vulkan doesn't look to good to me. I'd go for a function overload instead of having both in the same function.